start-mycroft.sh

```diff
@@ -14,15 +14,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-SOURCE="${BASH_SOURCE[0]}"
+# This script is never sourced but always directly executed, so this is safe to do
+SOURCE="$0"
```

Comment:
Don't know how I glossed over this previously but the script is sourced in the `bin/mycroft-start` helper command.

This means the $PWD will get set to `mycroft-core/bin` which causes issues in `check_dependencies` when it is reading from files stored in the parent `mycroft-core` directory.

Reply:
Hmm yes you're right. POSIX doesn't have an easy way to do this, this will get ugly...

Reply:
It seems it's just not possible to determine the path of a source script from within itself using just POSIX features...
https://stackoverflow.com/questions/29832037/how-to-get-script-directory-in-posix-sh/29835459#29835459

So, not sure how to resolve this. `pwd` seems to always return the directory in which the script is located (at least for me with Busybox, GNU Coreutils seem to do the same), no matter if it's sourced or not, but I can not find anything about that behaviour online so I'm not sure if we should rely on it. Otherwise it would be as easy as `SOURCE=$(pwd)`.

Reply:
Thinking out loud here... in the helper commands could we set `$BASH_SOURCE=$(pwd)` or an equivalent variable. Then test for it's existence in the start/stop scripts rather than assume it is there or not?

Reply:
What helper commands do you mean? `bin/mycroft-start`?

Reply:
Looked a bit on this and the helper commands in bin that uses start-mycroft.sh and stop-mycroft.sh seem to fail (when called from outside of the mycroft-core root folder). I wonder if this was what Kris meant. but I'm not sure.

I think this is due to the fact that the files are sourced and not executed as a script. for me it seems to work if instead of sourcing the start-mycroft.sh the script is executed.

For example in mycroft-start the change would be
```diff
diff --git a/bin/mycroft-start b/bin/mycroft-start
index 65723c2c37..bda0967048 100755
--- a/bin/mycroft-start
+++ b/bin/mycroft-start
@@ -18,4 +18,4 @@ SOURCE="${BASH_SOURCE[0]}"
 cd -P "$( dirname "$SOURCE" )"/..
 DIR="$( pwd )"
 
-. "$DIR/start-mycroft.sh" $@
+"$DIR/start-mycroft.sh" $@
```

Reply:
I've just gone ahead and made this depend on #2802, as that'll completely replace `bin/mycroft-start` for a new script anyway. Maybe at some point we can get rid of these location dependent scripts entirely with just `bin/mycroft-{start,stop}`, who knows.